### PR TITLE
Add ESIL flag update for LSL, LSR, ADC + correction on ADDS + test added

### PR DESCRIPTION
**Pull request by 0x8008135**
**Your checklist for this pull request**
- [*] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [*] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [*] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

Fixed BLS instruction in ESIL
Fixed ADC and ADCS instruction in ESIL
Fixed LSL/LSLS and LSR/LSRS instruction in ESIL
Added tests for the above
Fixed tests for STRLS for ARM32

...

**Test plan**
Tests for the above instructions have been updated accordingly

...

**Closing issues**
Probably :)
...

### CODE REVIEW

**Comment by trufae** [comment]
cc  @condret

**Comment by trufae** [comment]
cc @mrmacete

File: test/db/esil/arm_16 [hunk 1]
```diff
@@ -334,19 +334,51 @@ EXPECT=<<EOF
 EOF
 RUN
 
-NAME=lsrs r4, r4, 2
+NAME=lsls r1, r2
```
Comment (trufae) [inline test/db/esil/arm_16:337]: why remove the lsrs test?

File: libr/anal/p/anal_arm_cs.c [hunk 2]
```diff
@@ -1572,9 +1578,23 @@ static int analop_esil(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len
 		MATH32_NEG("|");
 		break;
 	case ARM_INS_LSR:
+	        if (insn->detail->arm.update_flags) {
+			if (OPCOUNT() == 2) {
+				r_strbuf_appendf (&op->esil, "%s,!,!,?{,%s,1,%s,-,0x1,<<,&,!,!,cf,:=,},",ARG(1),ARG(0),ARG(1));
```
Comment (trufae) [inline libr/anal/p/anal_arm_cs.c:1583]: missing space after , in the ARGs, in all the other places

File: test/db/esil/arm_16 (hunk 1 above)
Comment (0x8008135) [reply test/db/esil/arm_16:337]: In fact, the test is still present below starting at line 361
https://github.com/radareorg/radare2/blob/9957224a4dd5ef767b28df48790b3d698112eddc/test/db/esil/arm_16#L361-L383

The check for LSRS was adapted to test the update of the carry flag

File: libr/anal/p/anal_arm_cs.c (hunk 2 above)
Comment (condret) [inline libr/anal/p/anal_arm_cs.c:1583]: is the trailing comma needed here?

Comment (0x8008135) [reply libr/anal/p/anal_arm_cs.c:1583]: yes, as other flags need to be updated afterwards

lsrs r1, r2         r2,!,!,?{,r1,1,r2,-,0x1,<<,&,!,!,cf,:=,},r2,r1,>>,0xffffffff,&,r1,=,$z,zf,:=,31,$s,nf,:=      

lsrs update the N and Z flags according to the result

**Comment by ret2libc** [comment]
@trufae I guess this one can be merged, right?

**Comment by trufae** [comment]
i have approved it already, so it's good for me to press the big green button

**Pull request merged**